eskip/eskip.go

```diff
@@ -44,13 +53,28 @@ type parsedRoute struct {
 	backend  string
 }
 
+// A Predicate object represents a parsed, in-memory, route matching predicate
+// that is defined by extensions.
+type Predicate struct {
+
+	// The name of the custom predicate as referenced
+	// in the route definition. E.g. 'Foo'.
+	Name string
+
+	// The arguments of the predicate as defined in the
+	// route definition. The arguments can be of type
+	// float64 or string (string for both strings and
+	// regular expressions).
+	Args []interface{}
```

Comment:
this is the `Either` in scala... or a `float64 | string` in [Ceylon](http://ceylon-lang.org/documentation/1.2/reference/structure/type/) :)